include/blazesym.h

```diff
@@ -0,0 +1,519 @@
+#ifndef __blazesym_h_
+#define __blazesym_h_
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/**
+ * Feature names of looking up addresses of symbols.
+ */
+typedef enum blazesym_faf_type {
+  /**
+   * Invalid type
+   */
+  FAF_T_INVALID,
+  /**
+   * Return the offset in the file. (enable)
+   */
+  FAF_T_OFFSET_IN_FILE,
+  /**
+   * Return the file name of the shared object. (enable)
+   */
+  FAF_T_OBJ_FILE_NAME,
+  /**
+   * Return symbols having the given type. (sym_type)
+   */
+  FAF_T_SYMBOL_TYPE,
+} blazesym_faf_type;
+
+/**
+ * Names of the BlazeSym features.
+ */
+typedef enum blazesym_feature_name {
+  /**
+   * Enable or disable returning line numbers of addresses.
+   *
+   * Users should set `blazesym_feature.params.enable` to enabe or
+   * disable the feature,
+   */
+  LINE_NUMBER_INFO,
+  /**
+   * Enable or disable loading symbols from DWARF.
+   *
+   * Users should `blazesym_feature.params.enable` to enable or
+   * disable the feature.  This feature is disabled by default.
+   */
+  DEBUG_INFO_SYMBOLS,
+} blazesym_feature_name;
+
+/**
+ * Types of symbol sources and debug information for C API.
+ */
+typedef enum blazesym_src_type {
+  /**
+   * Symbols and debug information from an ELF file.
+   */
+  SRC_T_ELF,
+  /**
+   * Symbols and debug information from a kernel image and its kallsyms.
+   */
+  SRC_T_KERNEL,
+  /**
+   * Symbols and debug information from a process, including loaded object files.
+   */
+  SRC_T_PROCESS,
+} blazesym_src_type;
+
+/**
+ * The types of symbols.
+ *
+ * This type is used to choice what type of symbols you like to find
+ * and indicate the types of symbols found.
+ */
+typedef enum blazesym_sym_type {
+  /**
+   * Invalid type
+   */
+  SYM_T_INVALID,
+  /**
+   * You want to find a symbol of any type.
+   */
+  SYM_T_UNKNOWN,
+  /**
+   * The returned symbol is a function, or you want to find a function.
+   */
+  SYM_T_FUNC,
+  /**
+   * The returned symbol is a variable, or you want to find a variable.
+   */
+  SYM_T_VAR,
+} blazesym_sym_type;
+
+/**
+ * BlazeSymbolizer provides an interface to symbolize addresses with
+ * a list of symbol sources.
+ *
+ * Users should present BlazeSymbolizer with a list of symbol sources
+ * (`SymbolSrcCfg`); for example, an ELF file and its base address
+ * (`SymbolSrcCfg::Elf`), or a Linux kernel image and a copy of its
+ * kallsyms (`SymbolSrcCfg::Kernel`).  Additionally, BlazeSymbolizer
+ * uses information from these sources to symbolize addresses.
+ */
+typedef struct BlazeSymbolizer BlazeSymbolizer;
```

Comment:
does bindgen generate all these typedefs or it's somehow controllable? It's a bit out of style for kernel and libbpf styles, so I wonder if we can get rid of them?

Reply:
I haven't found a way to customize that aspect of the generation.